graphix/sim/density_matrix.py

```diff
@@ -74,10 +74,15 @@ def check_size_consistency(mat):
                 try:
                     if isinstance(input_list[0], Iterable) and isinstance(input_list[0][0], numbers.Number):
                         self.rho = np.array(input_list)
-                        assert check_square(self.rho)
+                        if not lv.is_square(self.rho):
+                            raise ValueError("Density matrix must be square.")
+                        if not lv.is_qubitop(self.rho):
+                            raise ValueError("Cannot interpret the provided density matrix as a qubit operator.")
```

Comment:
A function `ensure_qubitop(self.rho)` would remove code duplication and factor the logic of having dedicated error messages to catch not-square matrices between this method and `KrausData.__init__`.